docs/src/apps/programming-faq.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Programming FAQ"
+---
+
+When writing or interacting with Solana programs, there are common questions or challenges that often come up.  Below are resources to help answer these questions.  If not addressed here, Solana [#developers Discord channel](https://discord.gg/RxeGBH) is a great resource.
+
+
+## InvalidInstructionData
+
+This program error can occur while trying to deserialize the instruction, check
+that the structure passed in matches exactly the instruction.  There may be some
+padding between fields.  Try packing and unpacking the instruction type `T` to
+determine the exact encoding the program expects
+
+```rust
+fn unpack<T>(input: &[u8]) -> &T;
+fn pack<T>(input: &T) -> &[u8];
+```
+
+## InvalidAccountData
+
+This program error can happen for a lot of reasons, Usually, it's caused by
+passing an account to the program that the program is not expecting, either in
+the wrong position in the instruction or an account not compatible with the
+instruction being executed.
+
+An implementation of a program might also cause this error when performing a
+cross-program instruction and forgetting to provide the account for the program
+that you are calling.
+
+## MissingRequiredSignature
+
+Some instructions require the account to be a signer; this error is returned if an account expected to be signed is not.
+
+An implementation of a program might also cause this error when performing a cross-program invocation that requires a signed program address, but the passed signer seeds passed to `invoke_signed` don't match the signer seeds used to create the program address (`create_program_address`).
+
+## Using float types
+
+Solana programs support a limited subset of Rust's float operations, though they are highly discouraged due to the overhead involved.  If a program attempts to use a float operation that is not supported, the compiler/linker will report an unresolved symbol error.
```

Comment:
The unresolved symbol error is actually only thrown when running the program.  So this line can also encourage integration / system tests, something like "... the runtime will report an unresolved symbol error.  Be sure to include integration tests against a local cluster to ensure the operation is supported."